web/src/app/browser/src/defaultBrowserRules.ts

```diff
@@ -36,16 +36,22 @@ export default class DefaultBrowserRules extends DefaultRules {
 
     const contextManager = this.contextManager;
 
+    let elem: HTMLElement;
     switch(code) {
-      // This method will be transplanted to the specific `ContextManager` module stored above.
+      // This method will be handled between `ContextManager` and PageContextAttachment:
+      // pageContextAttachment.findNeighboringInput(contextManager.activeTarget.getElement(), <same flag>)
       case Codes.keyCodes['K_TAB']:
-        domManager.moveToNext((Lkc.Lmodifiers & Codes.modifierCodes['SHIFT']) != 0);
+        const bBack = (Lkc.Lmodifiers & Codes.modifierCodes['SHIFT']) != 0;
+        elem = contextManager.page.findNeighboringInput(contextManager.activeTarget.getElement(), bBack);
+        elem.focus();
         break;
       case Codes.keyCodes['K_TABBACK']:
-        domManager.moveToNext(true);
+        elem = contextManager.page.findNeighboringInput(contextManager.activeTarget.getElement(), true);
+        elem.focus();
         break;
       case Codes.keyCodes['K_TABFWD']:
-        domManager.moveToNext(false);
+        elem = contextManager.page.findNeighboringInput(contextManager.activeTarget.getElement(), false);
+        elem.focus();
         break;
     }
```

Comment:
This feels a lot WETter than before. Could we instead extract the `findNeighboringInput` call, something like:

```ts
    (/* function handle_tab */ () => {
      let bBack;
      switch(code) {
        // This method will be handled between `ContextManager` and PageContextAttachment:
        // pageContextAttachment.findNeighboringInput(contextManager.activeTarget.getElement(), <same flag>)
        case Codes.keyCodes['K_TAB']:
          bBack = (Lkc.Lmodifiers & Codes.modifierCodes['SHIFT']) != 0; 
          break;
        case Codes.keyCodes['K_TABBACK']: bBack = true; break;
        case Codes.keyCodes['K_TABFWD']:  bBack = false; break;
        default: 
          return;
      }
      let elem = contextManager.page.findNeighboringInput(contextManager.activeTarget.getElement(), bBack);
      elem.focus();
    })();
```

Or:

```ts
    // This method will be handled between `ContextManager` and PageContextAttachment:
    // pageContextAttachment.findNeighboringInput(contextManager.activeTarget.getElement(), <same flag>)
    const tabs = { 
      [Codes.keyCodes.K_TAB]: (Lkc.Lmodifiers & Codes.modifierCodes['SHIFT']) != 0, 
      [Codes.keyCodes.K_TABBACK]: true,
      [Codes.keyCodes.K_TABFWD]: false
    };

    const bBack = tabs[code];
    if(bBack !== undefined) {
      const elem = contextManager.page.findNeighboringInput(contextManager.activeTarget.getElement(), bBack);
      elem.focus();
    }
```

Or, perhaps this is neatest of all:

```ts
    const moveToNext = (back: boolean) => this.contextManager.page.findNeighboringInput(this.contextManager.activeTarget.getElement(), back).focus();
    switch(code) {
      // This method will be handled between `ContextManager` and PageContextAttachment:
      // pageContextAttachment.findNeighboringInput(contextManager.activeTarget.getElement(), <same flag>)
      case Codes.keyCodes.K_TAB:
        moveToNext((Lkc.Lmodifiers & Codes.modifierCodes.SHIFT) != 0);
        break;
      case Codes.keyCodes.K_TABBACK:
        moveToNext(true);
        break;
      case Codes.keyCodes.K_TABFWD:
        moveToNext(false);
        break;
    }
```